### ISSUE TITLE
Cannot create Client() due to lack of redirect URI

### ISSUE THREAD

**Issue by valknight** [qwertyquerty/pypresence#62]
**Issue**

Can't connect over RPC to access the Discord API, as data sent in Client.authorize() does not contain a URL to redirect to.

**Platform / Dependencies**

Ubuntu 18.04.2 LTS
Python 3.6.7
pypresence 3.3.1

**Steps to reproduce**

Run the following code

```
from pypresence import Client
client_id = "replaceme"
client = Client(client_id)
client.start()
client.authorize(client_id, scopes=["rpc", "rpc.api"]) # this throws a ServerError exception
```
This will cause the following error to be thrown:
```
pypresence.exceptions.ServerError: Oauth2 error: invalid_request: missing redirect uri.
```

Discord sends back the same error as above, with a status code of 5000.

**Attempts to fix**
I tried manually adding the data URL when being sent by modifying the BaseClient class, however the discord client still didn't like this, as I believe you need to manually modify the HTTP headers. This seemed a bit to wild for me at this point, as I didn't want to unintentionally break any other features without making an issue first.

**Comment by LewdNeko** [comment]
`manually modify the HTTP headers` ????
Are you alright there? Couple of things

a) When I tested it, it worked fine
b) Have you read https://github.com/qwertyquerty/pypresence/blob/master/examples/client-authorize-example.md ?
c) Do you have an understanding of OAuth. I assume so but if not, please make sure you do before handling things like this.

**Comment by valknight** [comment]
I actually have looked through the examples, hence opening an issue. Here's a screenshot of the output of that exact linked example, copy pasted with only the client ID modified if you don't believe me  - https://i.imgur.com/g4Zs6nd.png

If it helps, Discord does actually popup and give me the authenticate dialog, and upon clicking "Authenticate", it then drops me back here. *Also* before you give me shit for not doing it, yes I have added `localhost` and `127.0.0.1` as valid callbacks for OAuth, as should be required for something like this - screenshot of that https://i.imgur.com/uJ87enx.png

**Comment by LewdNeko** [comment]
That's great and all, but are you listening locally for requests to `localhost`. Or maybe you should setup a web server to listen & do something. By "drops me back here" what do you mean?

**Comment by valknight** [comment]
I have a server on localhost, but, no requests are being made to it. This is because Discord is refusing to send any data to it, because the request *to Discord* doesn't give a URL for me to redirect to, and I can't seem (unless  hence the server is useless.

Drops me back here means it gives me the error, after clicking authenticate (when you click the button, in the dev console discord shows the error being sent back)

To show you what I mean, and the entire flow and what it looks like, as well as Discord's log, I recorded a gif of the process https://i.imgur.com/tYFQhq8.gif - it shows the NGinx server listening for requests (note: no requests are made back to localhost *by example.py*, as, unless I've missed a core bit of your docs, you don't provide a URL to redirect to)

Again, I'm sure I'm missing something really easy here, but, I don't think it's a web server gone wrong, especially as the script is showing the exact error Discord is giving out, meaning the connection between Discord and the script *is* functioning, just the script isn't providing any callback URL, and, as far as I can tell, there's no way to specify what `client.authorize()`  should provide to Discord as a callback. I did a quick read of [Discord's documentation on the matter](https://discordapp.com/developers/docs/topics/rpc#connecting) with the following line sticking out to me

"If you're connecting to the RPC server from within a non-browser application (like a game), you just need to make sure that the origin is sent with the upgrade request when connecting to the WebSocket. For local testing, we recommend testing with an origin like https://localhost. For production apps, we recommend setting the origin to your company/game's domain, for example https://discordapp.com."

From what I can tell, the upgrade request isn't given (unless I'm massively misinterpreting Discord's logs), hence Discord throws an OAuth2 error.

If it's due to this library not supporting the WebSocket, that's fair enough, and I'm sorry for wasting your time. (Just, maybe add a note in future :sweat_smile:)
